### PR TITLE
Allow user to override <TAB> and <S-TAB> keys.

### PR DESCRIPTION
**Pull request by Traap**
Steps for submitting a pull request:

- [x] **ALL** pull requests should be made against the `dev` branch!
- [x] Take a look at [CONTRIBUTING.MD](https://github.com/vimwiki/vimwiki/blob/dev/CONTRIBUTING.md)
- [x] Reference any related issues.
- [x] Provide a description of the proposed changes.
- [x] PRs must pass Vint tests and add new Vader tests as applicable.
- [x] Make sure to update the documentation in `doc/vimwiki.txt` if applicable,
      including the Changelog and Contributors sections.

This pull request allows a vimwiki user to redefine the TAB and S-TAB keys.  The lines below are added to vimrc to override the default <TAB> and <S-TAB> behavior.
``  
let g:vimwiki_tab_key = '<F7>'
``
``
let g:vimwiki_shift_tab_key = '<F8>'
``

ftplugin/vimwiki.vim assigns these values to script variables when they exist, otherwise, TAB and S-TAB remain the defaults.

key mappings and insert mode mappings use g:vimwiki_tab_key and g:vimwiki_shift_tab_key to achieve the desired result.

### CODE REVIEW

**Comment by Traap** [comment]
I use the following development environment:

- Windows 10 Home 
- Version 2004
- OS Build 19041.330
- Experience Windows Feature Experience Pack 120.2202.130.0
- Windows Subsystem for Linux 2 (WSL2)
- Ubuntu LTS 20.04

I am not able to docker in this environment for various reasons.  I had to manually run the failing Vader tests.   All tests are now passing.

**Comment by tinmarino** [comment]
Hello,
Sorry for the delay : we add so many issues that I did not see the PRs.

May you add:
1/ Non regression tests
2/ Documentation

Without both, a new feature will be useless and annoying to maintain.

File: ftplugin/vimwiki.vim [hunk 1]
```diff
@@ -329,6 +329,15 @@ command! -buffer VimwikiCatUrl call vimwiki#html#CatUrl(expand('%:p'))
 " Keybindings
 " ------------------------------------------------
 
+" Allow the use to redefine <TAB> and <S-TAB>.
```
Comment (tinmarino) [inline ftplugin/vimwiki.vim:332]: PLesae use the default variable treatment, see in vars.vim or this commit: 97e02897f4d7e50c

File: doc/vimwiki.txt [hunk 2]
```diff
@@ -3595,6 +3595,7 @@ Contributors and their Github usernames in roughly chronological order:
     - flex (@bratekarate)
     - Marius Lopez (@PtitCaius)
     - Edward Bassett (@ebassett)
+    - Gary A. Howard (@Traap)
```
Comment (tinmarino) [inline doc/vimwiki.txt:3598]: Please document your variables
And for fast integration, make some test.

**Comment by tinmarino** [comment]
Please open a new PR when ready: it was the last pending one => too tempting to close :-)

**Pull request closed without merging**